Keen.NetStandard.Test/DatasetTests_Integration.cs

```diff
@@ -0,0 +1,434 @@
+using Keen.Core.Dataset;
+using Keen.Core.Query;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+
+namespace Keen.Core.Test
+{
+    [TestFixture]
+    public class DatasetTests_Integration : TestBase
+    {
+        private const string _datasetName = "video-view";
+        private const string _indexBy = "12";
+        private const string _timeframe = "this_12_days";
+
+
+        [Test]
+        public void Results_Success()
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/GetDatasetResults.json");
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForGetAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var dataset = client.QueryDataset(_datasetName, _indexBy, _timeframe);
+
+            Assert.IsNotNull(dataset);
+            Assert.IsNotNull(dataset["result"]);
+        }
+
+        [Test]
+        public void Definition_Success()
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/GetDatasetDefinition.json");
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForGetAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var dataset = client.GetDatasetDefinition(_datasetName);
+
+            AssertDatasetIsPopulated(dataset);
+        }
+
+        [Test]
+        public void ListDefinitions_Success()
+        {
+            var apiResponse = File.ReadAllText($"{GetApiResponsesPath()}/ListDatasetDefinitions.json");
+            IKeenHttpClientProvider httpClientProvider = null;
+
+            if (UseMocks)
+            {
+                httpClientProvider = GetMockHttpClientProviderForGetAsync(apiResponse);
+            }
+
+            var client = new KeenClient(SettingsEnv, httpClientProvider);
+            var datasetCollection = client.ListDatasetDefinitions();
+
+            Assert.IsNotNull(datasetCollection);
+            Assert.IsNotNull(datasetCollection.Datasets);
+            Assert.IsTrue(datasetCollection.Datasets.Any());
+            Assert.IsTrue(!string.IsNullOrWhiteSpace(datasetCollection.NextPageUrl));
```

Comment:
Again, would be great to verification of values instead of just checking to see that anything is there.

Reply:
Added a blurb to Issue #130 about this.